server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -658,6 +658,9 @@ public void getLocalShuffleData(
                 .setRetMsg(msg)
                 .build();
       } finally {
+        if (sdr != null) {
```

Comment:
Maybe we should release `ShuffleIndexResult`, too.

Reply:
@jerqi 
In fact, getLocalShuffleData and getLocalShuffleIndex only use FileSegmentManagedBuffer. It means will trigger direct  memory leak. I fix this only to keep pace with getMemoryShuffleData.
Do you think we should fix ShuffleIndexResult to keep pace with this? Or remove the fix on getLocalShuffleData?

Reply:
Yes, we can remove the fix on `getLocalShuffleData` and add some comments to explain why we don't need the fix. Or we fix the method `getLocalShuffleIndex`, too.

Reply:
> Yes, we can remove the fix on `getLocalShuffleData` and add some comments to explain why we don't need the fix. Or we fix the method `getLocalShuffleIndex`, too.

Ok, I will fix the method `getLocalShuffleIndex`.
For now, even though FileSegmentManagedBuffer::release do nothing,  I am afraid that allocate some resource in the future.